File: backend/app/models/doctor.py
from enum import Enum

from sqlalchemy import (
    Column, Integer, String, Text, Float, Boolean,
    ForeignKey, DateTime, func, Enum as SqlEnum,
)
from sqlalchemy.orm import relationship

from app.models.base import Base


class Specialization(str, Enum):
    CARDIOLOGY = "Cardiology"
    DERMATOLOGY = "Dermatology"
    NEUROLOGY = "Neurology"
    PEDIATRICS = "Pediatrics"
    ORTHOPEDICS = "Orthopedics"
    GYNECOLOGY = "Gynecology"
    PSYCHIATRY = "Psychiatry"
    ONCOLOGY = "Oncology"
    RADIOLOGY = "Radiology"
    UROLOGY = "Urology"
    GASTROENTEROLOGY = "Gastroenterology"
    GENERAL_PHYSICIAN = "General Physician"


class Department(str, Enum):
    MEDICINE = "Medicine"
    SURGERY = "Surgery"
    EMERGENCY = "Emergency"
    ICU = "ICU"
    ENT = "ENT"
    OPD = "OPD"
    RADIOLOGY = "Radiology"
    ORTHOPEDICS = "Orthopedics"


class DoctorStatus(str, Enum):
    PENDING = "pending"
    APPROVED = "approved"
    REJECTED = "rejected"


def _values(obj):
    return [e.value for e in obj]


class Doctor(Base):
    __tablename__ = "doctors"

    id = Column(Integer, primary_key=True, index=True)
    doctor_id = Column(String(20), unique=True, nullable=False, index=True)
    full_name = Column(String(100), nullable=False)

    specialization = Column(
        SqlEnum(Specialization, name="doctor_specialization", values_callable=_values),
        nullable=False,
    )
    department = Column(
        SqlEnum(Department, name="doctor_department", values_callable=_values),
        nullable=False,
    )

    qualification = Column(String(150), nullable=False)
    experience = Column(Integer, nullable=False)
    consultation_fee = Column(Float, nullable=False)
    availability = Column(Text, nullable=True)

    phone_number = Column(String(15), unique=True, nullable=False)
    email = Column(String(100), unique=True, nullable=True)
    address = Column(Text, nullable=True)

    is_active = Column(Boolean, default=False, nullable=False)

    status = Column(
        SqlEnum(DoctorStatus, name="doctor_status", values_callable=_values),
        default=DoctorStatus.PENDING,
        nullable=False,
    )

    created_by = Column(Integer, ForeignKey("users.id"), nullable=True)
    created_by_user = relationship("User", foreign_keys=[created_by])

    user_id = Column(Integer, ForeignKey("users.id"), nullable=True, unique=True)
    user = relationship("User", foreign_keys=[user_id], backref="doctor_profile")

    created_at = Column(DateTime, nullable=False, server_default=func.now())
    updated_at = Column(
        DateTime, nullable=False,
        server_default=func.now(), onupdate=func.now(),
    )

File: frontend /src/types/doctor.ts
export type DoctorStatus = 'pending' | 'approved' | 'rejected';

export type Specialization =
  | 'Cardiology' | 'Dermatology' | 'Neurology' | 'Pediatrics'
  | 'Orthopedics' | 'Gynecology' | 'Psychiatry' | 'Oncology'
  | 'Radiology' | 'Urology' | 'Gastroenterology' | 'General Physician';

export type Department =
  | 'Medicine' | 'Surgery' | 'Emergency' | 'ICU'
  | 'ENT' | 'OPD' | 'Radiology' | 'Orthopedics';

export interface Doctor {
  id: number;
  doctor_id: string;                     // e.g. "DOC00000001"
  full_name: string;
  specialization: Specialization;
  department: Department;
  qualification: string;
  experience: number;
  consultation_fee: number;
  availability?: string | null;
  phone_number: string;
  email?: string | null;
  address?: string | null;
  is_active: boolean;
  status: DoctorStatus;
  created_by?: number | null;
  created_at: string;
  updated_at: string;
}

export interface DoctorRegistrationRequest {
  full_name: string;
  specialization: Specialization;
  department: Department;
  qualification: string;
  experience: number;
  consultation_fee: number;
  availability?: string;
  phone_number: string;
  email: string;                         // required per backend fix
  password: string;
  address?: string;
}

export type DoctorUpdateRequest = Partial
  Omit<DoctorRegistrationRequest, 'password' | 'email'>
> & { email?: string };

export interface DoctorFilters {
  search?: string;
  department?: Department;
  specialization?: Specialization;
  status?: DoctorStatus;
  page?: number;
  limit?: number;
}